gto-support-lifecycle/src/main/java/org/ccci/gto/android/common/lifecycle/Transformations2.kt

```diff
@@ -0,0 +1,30 @@
+@file:JvmName("Transformations2")
+
+package org.ccci.gto.android.common.lifecycle
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MediatorLiveData
+import androidx.lifecycle.Observer
+
+/**
+ * This method will combine 2 LiveData objects into a new LiveData object by running the {@param mapFunction} on the
+ * current values of both source LiveData objects.
+ * Returning either of the original LiveData objects will cause an Exception.
+ */
+@JvmName("combine")
+fun <X, Y, Z> LiveData<X>.combineWith(other: LiveData<Y>, mapFunction: (X?, Y?) -> LiveData<Z>): LiveData<Z> {
+    val result = MediatorLiveData<Z>()
```

Comment:
Can this be used to combine two LiveData calls?

Reply:
depends on what you mean by that, this is based off of `Transformations.switchMap()`

Reply:
Ok, so not what I was thinking of.  I was thinking more on the lines of an `Observable.zip()` replacement.

Reply:
you could utilize it for something like that. I had trouble figuring out a good name for this method. As it stands I'm not really sold on the `combine`/`combineWith` name, but I don't know of anything better.